defaults/initrd.scripts

```diff
@@ -2055,7 +2139,11 @@ openLUKS() {
 						[ -e /dev/tty ] && run mv /dev/tty /dev/tty.org
 						run mknod /dev/tty c 5 1
 						cryptsetup_options="${cryptsetup_options} -d -"
-						gpg_cmd="gpg --logger-file /dev/null --quiet --decrypt ${mntkey}${LUKS_KEY} |"
+						gpg_cmd="gpg --logger-file /dev/null --quiet --decrypt ${mntkey}${LUKS_KEY} | "
+						# plymouth password entry is passed through STDIN, requiring '--passphrase-fd 0 --batch'
+						# for newer gpg versions (>=2.1) '--pinentry-mode loopback' may also be required for the above
+						# '--no-tty' is included to prevent interruption of plymouth by any output
+						ply_cmd="gpg --logger-file /dev/null --quiet --passphrase-fd 0 --batch --no-tty --decrypt ${mntkey}${LUKS_KEY} | "
```

Comment:
```
gpg_cmd="gpg --logger-file /dev/null --quiet --decrypt ${mntkey}${LUKS_KEY}"

if [ "${PLYMOUTH}" = '1' -a "${CRYPT_SILENT}" != '1' ]
then
	gpg_cmd="${gpg_cmd} --passphrase-fd 0 --batch --no-tty |"
else
	gpg_cmd="${gpg_cmd} |"
fi
```

Reply:
Reread the gpg man page, looks like this should work fine.  Added!

Reply:
I like how this reduces duplication, but it causes issues when the '--decrypt' option isn't last.  I've implemented a slight variation that works.